ods_ci/tasks/Resources/Provisioning/GPU/amd_operator/amd_operator.sh

```diff
@@ -56,35 +56,44 @@ has_csv_succeeded() {
 }
 
 function create_devconfig() {
-  oc create -f - <<EOF
+  dc_name="dc-internal-registry"
+  dc=$(oc get DeviceConfig $dc_name -n openshift-amd-gpu -oname --ignore-not-found)
+  if [[ -n $dc ]];
+    then
+      echo "AMD DeviceConfig $dc_name already exists". Skipping creation
+    else
+      echo "Creating AMD DeviceConfig..."
+      oc create -f - <<EOF
 kind: DeviceConfig
 apiVersion: amd.io/v1alpha1
 metadata:
-  name: dc-internal-registry
+  name: $dc_name
   namespace: openshift-amd-gpu
 EOF
+  fi
 }
 
-function wait_until_pod_ready_status() {
-  local timeout_seconds=1200
-  local pod_label=$1
-  local namespace=$2
-  local timeout=240
+
+function wait_until_pod_is_created() {
+  label=$1
+  namespace=$2
+  timeout=$3
   start_time=$(date +%s)
   while [ $(($(date +%s) - start_time)) -lt $timeout ]; do
-     pod_status="$(oc get pod -l app="$pod_label" -n "$namespace" --no-headers=true 2>/dev/null)"
-     daemon_status="$(oc get daemonset -l app="$pod_label" -n "$namespace" --no-headers=true 2>/dev/null)"
-     if [[ -n "$daemon_status" || -n "$pod_status" ]] ; then
-        echo "Waiting until GPU Pods or Daemonset of '$pod_label' in namespace '$namespace' are in running state..."
-        echo "Pods status: '$pod_status'"
-        echo "Daemonset status: '$daemon_status'"
-        oc wait --timeout="${timeout_seconds}s" --for=condition=ready pod -n "$namespace" -l app="$pod_label" || \
-        oc rollout status --watch --timeout=3m daemonset -n "$namespace" -l app="$pod_label" || continue
+    podName=$(oc get pods -n $2 -l $1 -oname)
+    if [[ -n $podName ]];
+      then {
+        echo Pod $podName found!
+        return 0
```

Comment:
when you add `return` in this line the `break` in the next line IMH is dead code and it will be better to remove it

Reply:
true, it's a residual from previous changes. Will remove the break thanks

Reply:
done